R/config.R

```diff
@@ -1,45 +1,126 @@
-check_yaml_has <- function(missing, where, pkg, call = caller_env()) {
-  if (length(missing) == 0) {
-    return()
-  }
+config_pluck <- function(pkg, path, default = NULL) {
+  check_string(path, allow_empty = FALSE, .internal = TRUE)
 
-  missing_components <- lapply(missing, function(x) c(where, x))
-  msg_flds <- purrr::map_chr(missing_components, paste, collapse = ".")
+  where <- strsplit(path, ".", fixed = TRUE)[[1]]
+  purrr::pluck(pkg$meta, !!!where, .default = default)
+}
 
-  config_abort(
-    pkg, 
-    "Can't find {cli::qty(missing)} component{?s} {.field {msg_flds}}.",
-    call = call
+config_pluck_character <- function(pkg,
+                                   path,
+                                   default = character(),
+                                   call = caller_env()) {
+  x <- config_pluck(pkg, path, default)
+  config_check_character(
+    x,
+    error_path = path,
+    error_pkg = pkg,
+    error_call = call
   )
 }
 
-yaml_character <- function(pkg, where) {
-  x <- purrr::pluck(pkg$meta, !!!where)
+config_pluck_string <- function(pkg,
+                                path,
+                                default = "",
+                                call = caller_env()) {
+  x <- config_pluck(pkg, path, default)
+  config_check_string(
+    x,
+    error_path = path,
+    error_pkg = pkg,
+    error_call = call
+  )
+}
+
+# checks ---------------------------------------------------------------------
 
-  if (identical(x, list()) || is.null(x)) {
+config_check_character <- function(x,
+                                   error_pkg,
+                                   error_path,
+                                   error_call = caller_env()) {
+  if (is.character(x)) {
+    x
+  } else if (identical(x, list())) {
     character()
```

Comment:
what is this case?

Reply:
I don't know 😄 But I think it must be some yaml special case where we want to handle an empty list.